File: interpreter/java/AllFood.java
package interpreter.java;

public class AllFood implements Item {

    private Item drinkItem;
    private Item foodItem;

    public AllFood(Item drinkItem, Item foodItem) {
        this.drinkItem = drinkItem;
        this.foodItem = foodItem;
    }

    @Override
    public void interpret(FoodOrder foodOrder) {
        drinkItem.interpret(foodOrder);
        foodItem.interpret(foodOrder);
    }
}

File: interpreter/golang/allfood.go
package main

// AllFood is a Non-Terminal Expression (AND logic)
type AllFood struct {
	Items []Item
}

func (af *AllFood) Interpret(order *FoodOrder) bool {
	for _, item := range af.Items {
		if !item.Interpret(order) {
			return false
		}
	}
	return true
}
